WMF Framework/Event Platform Client/EPCNetworkManager.swift

```diff
@@ -0,0 +1,135 @@
+
```

Comment:
If the below changes are able to be made to EPC/EPCNetworkManager, and the extraneous accessors are wrapped up in 3-4 functions for thread-safe arrays, then there will be a lot of room to put the remaining `schedule`/`send`/`sendAllScheduled` and the `fetchStreamConfiguration` in the main class. The truly implementation-dependent functions are `httpPost` and `httpGet` (see review below for what these are), but those are small. 

The reason I'm after this is because I think that's what this code "wants" -- by which I mean we are duplicating state without needing to (e.g. the DispatchQueue), encapsulating functions that are only ever called once (e.g. httpDownload), and instantiating without much reason (Session is already instantiated and Session.shared already exists). We are also not cleanly separating the truly implementation-dependent parts (e.g. the issuing of the HTTP requests) from the other parts of the biz logic. 

Unification would allow us to remove the protocol file as well, bringing us down to just the one file with a section for implementation-dependent threadsafe array access and implementation-dependent http get/post requests, and nothing else. This too makes sense to me since I don't think we plan to re-implement these components wth different logic but the same protocol/contract.

I'm happy to prepare a patchset if it would help

Reply:
This sounds good to me. My only ask is to lean on putting the network methods in a private extension on EPC, similar to how we put all of it's thread safety methods in a private extension.

Reply:
Yeah totally agree about putting them in a private extension. Whatever minimally opinionated structure we can grab that keeps us from accidentally coupling to the main class. Private extensions don't exactly prevent us from two-way coupling but I it works for me, all we need is a reminder for now that these are distinct. The interface will formalize over the next couple revs as things grow/evolve and it'll tell us what it wants to look like.

Reply:
Just pushed a commit where this was done.